### PR TITLE
[KEYCLOAK-19310] - Oracle Support

### PR DESCRIPTION
**Pull request by pedroigor**
@DGuhr @stianst We are now very strict about some key database settings such as Hibernate dialect, JDBC Driver, etc.

The versions we are enabling are pretty much based on our list of supported configurations (RH-SSO). I see that as a good thing because we are now more opinionated.

But yeah, people using databases that need different settings might be impacted.

### CODE REVIEW

**Comment by stianst** [comment]
@pedroigor could you elaborate a bit? Do you mean that a particular `vendor` sets the JDBC driver + Dialect, or other things as well? I think that's the right approach as we don't want users to have to deal with that.

**Comment by DGuhr** [comment]
+1 from my side to stians question. I think it's ok as you did, but I'm not entirely sure about the implications of setting driver/dialect statically has.iirc the dialect names are not a version pinning anymore (at least not for mssql/oracle), but "at least use oracle12c/mssql2012" so it should be ok in terms of maintenance, as long as they don't break this backwards compatibility.

**Comment by pedroigor** [comment]
Exactly. The settings we are using are also based on the configuration we support in RH-SSO.

Depending on the feedback we have, we can review this anytime.

**Review by stianst — APPROVED** [review]
LGTM, but @DGuhr should review before merging

**Review by DGuhr — APPROVED** [review]

**Comment by DGuhr** [comment]
LGTM :)

**Pull request merged**
